mod-source-record-manager-server/src/main/java/org/folio/services/afterprocessing/AdditionalFieldsUtil.java

```diff
@@ -262,43 +294,57 @@ public static String getValue(Record record, String tag, char subfield) {
    * remove field from marc record
    *
    * @param record record that needs to be updated
-   * @param field  tag of the field
+   * @param field tag of the field
    * @return true if succeeded, false otherwise
    */
   public static boolean removeField(Record record, String field) {
     boolean result = false;
     try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
-      if (record != null && record.getParsedRecord() != null && record.getParsedRecord().getContent() != null) {
-        MarcReader reader = buildMarcReader(record);
+      org.marc4j.marc.Record marcRecord = computeMarcRecord(record);
+      if (marcRecord != null) {
         MarcWriter marcStreamWriter = new MarcStreamWriter(new ByteArrayOutputStream());
         MarcJsonWriter marcJsonWriter = new MarcJsonWriter(baos);
-        if (reader.hasNext()) {
-          org.marc4j.marc.Record marcRecord = reader.next();
-          VariableField variableField = marcRecord.getVariableField(field);
-          if (variableField != null) {
-            marcRecord.removeVariableField(variableField);
-          }
-          // use stream writer to recalculate leader
-          marcStreamWriter.write(marcRecord);
-          marcJsonWriter.write(marcRecord);
-          record.setParsedRecord(record.getParsedRecord().withContent(new JsonObject(new String(baos.toByteArray())).encode()));
-          result = true;
+        VariableField variableField = marcRecord.getVariableField(field);
+        if (variableField != null) {
+          marcRecord.removeVariableField(variableField);
         }
+        // use stream writer to recalculate leader
+        marcStreamWriter.write(marcRecord);
+        marcJsonWriter.write(marcRecord);
+        record.setParsedRecord(
+            record
+                .getParsedRecord()
+                .withContent(new JsonObject(new String(baos.toByteArray())).encode()));
+        result = true;
       }
     } catch (Exception e) {
       LOGGER.error("Failed to remove controlled field {} from record {}", field, record.getId(), e);
     }
     return result;
   }
 
-  private static MarcReader buildMarcReader(Record record) {
-    return new MarcJsonReader(new ByteArrayInputStream(record.getParsedRecord().getContent().toString().getBytes(StandardCharsets.UTF_8)));
+  /**
+   * Generate a {@link org.marc4j.marc.Record} from {@link Record} passed in.
+   * Will return null when there is no parsed content string present. Generated MARC record will be saved into cache if
+   * its parsed content string is not present in the cache as a key
+   */
+  private static org.marc4j.marc.Record computeMarcRecord(Record record) {
+    if (record != null
+        && record.getParsedRecord() != null
+        && !StringUtils.isBlank(record.getParsedRecord().getContent().toString())) {
```

Comment:
We can simplify like this: record.getParsedRecord().getContent() != null

Reply:
That is what i had prior. I wrote a test where the string was `""` and it failed so i made it more robust.